astra-sim/system/Sys.cc

```diff
@@ -161,7 +161,7 @@ Sys::Sys(
   this->roofline = nullptr;
 
   this->mem = mem;
-  this->mem->set_sys(id, this);
+ // this->mem->set_sys(id, this);
```

Comment:
We want to enable memory models as needed. If this->mem is null, this line should not be executed. Otherwise, it's essential for the memory models to function properly.

Reply:
Hi @TaekyungHeo , thanks for taking a look. It seems the changes are not great to be compatible with and without memory model. My goal is making AstraSim to work with network simulator, where it may not provides the memory model. Thanks!